code/datums/components/religious_tool.dm

```diff
@@ -0,0 +1,149 @@
+	var/datum/religion_sect/easy_access_sect
+	/// What extent do we want this religious tool to act? In case you don't want full access to the list. Generated on New
+	var/operation_flags
+	/// The rite currently being invoked
+	var/datum/religion_rites/performing_rite
+	///Sets the type for catalyst
+	var/catalyst_type = /obj/item/storage/book/bible
+	///Enables overide of COMPONENT_NO_AFTERATTACK, not recommended as it means you can potentially cause damage to the item using the catalyst.
+	var/force_catalyst_afterattack = FALSE
+
+/datum/component/religious_tool/Initialize(_flags = ALL, _force_catalyst_afterattack = FALSE, override_catalyst_type)
+	. = ..()
+	SetGlobalToLocal() //attempt to connect on start in case one already exists!
+	operation_flags = _flags
+	force_catalyst_afterattack = _force_catalyst_afterattack
+	if(override_catalyst_type)
+		catalyst_type = override_catalyst_type
+
+/datum/component/religious_tool/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_RELIGION_NEWSECT, .proc/SetGlobalToLocal)
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY,.proc/AttemptActions)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/component/religious_tool/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_PARENT_ATTACKBY,COMSIG_RELIGION_NEWSECT, COMSIG_PARENT_EXAMINE))
+
+/**
+  * Sets the easy access variable to the global if it exists.
+  */
+/datum/component/religious_tool/proc/SetGlobalToLocal()
+	if(!GLOB.religious_sect)
+		return
+	easy_access_sect = GLOB.religious_sect
+
+
+/**
+  * Since all of these involve attackby, we require mega proc. Handles Invocation, Sacrificing, And Selection of Sects.
+  */
+/datum/component/religious_tool/proc/AttemptActions(datum/source, obj/item/the_item, mob/living/user)
+	/**********Sect Selection**********/
+	if(!easy_access_sect)
+		if(!(operation_flags & RELIGION_TOOL_SECTSELECT))
+			return
+		. = COMPONENT_NO_AFTERATTACK //At this point you're intentionally trying to select a sect.
+		if(user.mind.holy_role != HOLY_ROLE_HIGHPRIEST)
+			to_chat(user, "<span class='warning'>You are not the high priest, and therefore cannot select a religious sect.")
+			return
+		var/list/available_options = generate_available_sects(user)
+		if(!available_options)
+			return
+
+		var/sect_select = input(user,"Select a sect (You CANNOT revert this decision!)","Select a Sect",null) in available_options
+		if(!sect_select || !user.canUseTopic(parent, BE_CLOSE, FALSE, NO_TK))
+			to_chat(user,"<span class ='warning'>You cannot select a sect at this time.</span>")
+			return
+		var/type_selected = available_options[sect_select]
+		GLOB.religious_sect = new type_selected()
+		for(var/i in GLOB.player_list)
+			if(!isliving(i))
+				continue
+			var/mob/living/am_i_holy_living = i
+			if(!am_i_holy_living.mind?.holy_role)
+				continue
+			GLOB.religious_sect.on_conversion(am_i_holy_living)
+		easy_access_sect = GLOB.religious_sect
+		SEND_SIGNAL(parent, COMSIG_RELIGION_NEWSECT)
```

Comment:
I just realized what you're trying to do here and it's not going to work. This will send this signal to the parent which will call SetGlobalToLocal() which is pointless because you set easy_access_sect in the line above. any tools that don't have early_access_sect set will still have them as null because this signal is only being sent to this tool.